storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -78,33 +78,41 @@ func (s *peersSource) Read(
 		return result, nil
 	}
 
-	s.log.Infof("peers bootstrapper starting to bootstrap shards in parallel")
 	var (
-		bopts = s.opts.ResultOptions()
-		lock  sync.Mutex
-		wg    sync.WaitGroup
+		bopts       = s.opts.ResultOptions()
+		concurrency = s.opts.BootstrapShardConcurrency()
+		count       = len(shardsTimeRanges)
+		lock        sync.Mutex
+		wg          sync.WaitGroup
 	)
+	s.log.WithFields(
+		xlog.NewLogField("shards", count),
+		xlog.NewLogField("concurrency", concurrency),
+	).Infof("peers bootstrapper bootstrapping shards for ranges")
+	workers := xsync.NewWorkerPool(concurrency)
+	workers.Init()
 	for shard, ranges := range shardsTimeRanges {
-		it := ranges.Iter()
-		for it.Next() {
-			currRange := it.Value()
-			// We fetch all results in parallel and the underlying session will
-			// rate limit the total throughput with a single worker pool that
-			// is defined by the fetch series blocks batch concurrency option
-			wg.Add(1)
-			go func(shard uint32, start, end time.Time) {
+		shard, ranges := shard, ranges
```

Comment:
Yeah, should remove this now that we don't need to capture it for the goroutine that used to encapsulate the inner logic.